frame/contracts/src/exec.rs

```diff
@@ -1328,14 +1329,19 @@ where
 		&mut self.top_frame_mut().nested_gas
 	}
 
-	fn append_debug_buffer(&mut self, msg: &str) -> bool {
+	fn append_debug_buffer(&mut self, msg: &str) -> Result<bool, ()> {
 		if let Some(buffer) = &mut self.debug_message {
 			if !msg.is_empty() {
+				ensure!(
+					buffer.len().checked_add(msg.len()).ok_or(())? <=
+						self.schedule.limits.debug_buffer_len as usize,
+					()
+				);
```

Comment:
Use a `BoundedVec` for the debug buffer. Then we don't need custom logic. This will also require you to put the maximum debug buffer length directly into the `Config` trait instead of into the `Schedule`.